### DISCUSSION TITLE
not able to import useFocusRef hook

### DISCUSSION THREAD

**Discussion by iamravisingh** [Comcast/react-data-grid#2898] [Q&A]
Is there any way to import **useFocusRef** from react-data-grid? It seems it is not exported. Since it is not exported we are not able to use the column filter in our project. Quite tough to integrate what react-data-grid is giving. 
Any help is appreciated. Thanks in Advance.

Below is the filter component which I am trying to use.

```
const FilterContext = createContext<Filter | undefined>(undefined);
export const FilterRenderer = ({
    isCellSelected,
    column,
    children
  }) => {
    const filters = useContext(FilterContext)!;
    const { ref, tabIndex } = useFocusRef(isCellSelected);
  
    return (
      <>
        <div>{column.name}</div>
        {filters.enabled && <div>{children({ ref, tabIndex, filters })}</div>}
      </>
    );
  }
```

**Comment by TeeWallz** [comment]
I agree, the example code isn't great for end users as relies on the unbundled code to import hooks. Examples assume the code is in the package itself.

For example: [HeaderFilters.tsx](https://github.com/adazzle/react-data-grid/blob/main/website/demos/HeaderFilters.tsx) imports useFocusRef via
`import { useFocusRef } from '../../src/hooks';`

Which can't be done when importing the module.

Could the examples and code be written so that can be used in the field and outside of the unbundled repo?

Love the project by the way, thanks for the effort
